### DISCUSSION TITLE
Half-steps for Foreground scale?

### DISCUSSION THREAD

**Discussion by diverswan9** [nagadomi/nunif#211] [General]
Any chance to have Half-steps for Foreground scale? i.e. - '-0.5' etc.

**Comment by nagadomi** [comment]
I added support for specifying a float value for --foreground-scale option.

Note that foreground-scale is not a linear parameter (This means that foreground-scale 2 is not twice as effective as foreground-scale 1).
> Currently, foreground-scale is not a continuous value, so it is not possible to select anything other than 0,1,2,3.
> https://github.com/nagadomi/nunif/discussions/107#discussioncomment-8969147

In the current implementation,  when --foreground-scale is a float value, interpolate the two mappers.
For example,
`--foreground-scale 1.2` = `--foreground-scale 1` * 0.8 + `--foreground-scale 2` * 0.2

**Reply by nagadomi** [reply]
I am more concerned about 3D space distortion than depth resolution so I would like to improve mapper related features in the future.

**Reply by diverswan9** [reply]
Thanks! It works well... I sometimes have been using 'FS-1' and found it to be too much so somewhere in between gives more depth to the background than with a '0' setting. I compared fs0, fs-0.5, and fs-1. The difference is subtle which is good.

**Comment by andy500** [comment]
l think every one using your program  will be happy you are try to improve 3D space distortion  as one who using your program l well happy and may l say your program is one of best program that is free and amazing your conversation speed is incredible l can do a 4k 20min clip in 
35mins in 3d sbs high quality  using rtx 6000 gpu thank you for this program  nagadomi
